### PR TITLE
Vines in caves

### PR DESCRIPTION
**Pull request by NiLSPACE**
Added a vines finisher that is able to create vines in caves. It ranges from the surface to a configurable level.

### CODE REVIEW

**Comment by SphinxC0re** [comment]
Nice one :)

File: src/Generating/FinishGen.cpp [hunk 1]
```diff
@@ -244,6 +244,100 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
...
+		for (int z = 0; z < cChunkDef::Width; z++)
+		{
+			int zz = z + a_ChunkDesc.GetChunkZ() * cChunkDef::Width;
+			if (!IsJungleVariant(a_ChunkDesc.GetBiome(x, z)))
+			{
+				// Current biome isn't a jungle
+				continue;
+			}
+
+			if (m_Noise.IntNoise2D(xx, zz) < 0.5)
```
Comment (madmaxoft) [inline src/Generating/FinishGen.cpp]: Prefer the `IntNoise2DInt()` function to `IntNoise2D()`, it is a bit faster.
```cpp
if ((m_Noise.IntNoise2DInt(xx, zz) % 101) < 50)
```

File: src/Generating/ComposableGenerator.cpp [hunk 2]
```diff
@@ -616,6 +616,11 @@ void cComposableGenerator::InitFinishGens(cIniFile & a_IniFile)
 			int MaxDensity = a_IniFile.GetValueSetI("Generator", "VillageMaxDensity", 80);
 			m_FinishGens.push_back(std::make_shared<cVillageGen>(Seed, GridSize, MaxOffset, MaxDepth, MaxSize, MinDensity, MaxDensity, m_BiomeGen, m_CompositedHeightCache));
 		}
+		else if (NoCaseCompare(*itr, "Vines") == 0)
+		{
+			int Level = a_IniFile.GetValueSetI("Generator", "VinesLevel", 40);
+			m_FinishGens.push_back(cFinishGenPtr(new cFinishGenVines(Seed, Level)));
```
Comment (madmaxoft) [inline src/Generating/ComposableGenerator.cpp]: The `std::make_shared<>()` function is preferred.

**Comment by NiLSPACE** [comment]
I've fixed the issues

**Pull request merged**